app/controllers/scim_rails/scim_groups_controller.rb

```diff
@@ -45,5 +73,67 @@ def patch_update
 
     private
 
+    def put_error_check
+      members = params["members"]
+
+      raise ScimRails::ExceptionHandler::InvalidPutMembers unless (members.is_a?(Array) && array_of_hashes?(members))
+
+      member_ids = members.map{ |member| member["value"] }
+
+      member_ids.each do |id|
+        @company.public_send(ScimRails.config.scim_users_scope).find(id)
+      end
+        
+      return if put_active_param.nil?
+
+      case put_active_param
+      when true, "true", 1
+        
+      when false, "false", 0
+        
+      else
+        raise ScimRails::ExceptionHandler::InvalidActiveParam
+      end
+    end
+
+    def array_of_hashes?(array)
+      array.each do |element|
+        return false unless element.is_a?(Hash)
+      end
+
+      return true
+    end
+
+    def permitted_group_params(parameters)
+      ScimRails.config.mutable_group_attributes.each.with_object({}) do |attribute, hash|
+        hash[attribute] = parameters.dig(*group_path_for(attribute))
+      end
+    end
+
+    def update_group_status(group)
+      group.public_send(ScimRails.config.group_reprovision_method) if active?
+      group.public_send(ScimRails.config.group_deprovision_method) unless active?
+    end
+
+    def group_path_for(attribute, object = ScimRails.config.mutable_group_attributes_schema, path = [])
+      at_path = path.empty? ? object : object.dig(*path)
+      return path if at_path == attribute
+
+      case at_path
+      when Hash
+        at_path.each do |key, value|
+          found_path = path_for(attribute, object, [*path, key])
+          return found_path if found_path
+        end
+        nil
```

Comment:
Wondering about this instead.
There's no need to have nil in these lines because they should be what gets returned if nothing else does anyway.
Reduce here will return the first thing that exists
```suggestion
        at_path.reduce do |acc, pair|
          key, _value = pair
          acc || path_for(attribute, object, [*path, key])
        end
```